### ISSUE TITLE
`thumbnail: true` is not sufficient to get proper Thumbnail tag in custom builds

### ISSUE THREAD

**Issue by Jonarod** [mattiasw/ExifReader#425]
### Description

Including `thumbnail: true` in the custom build settings (in the package.json) is not enough to actually get the Thumbnail tag.

**Additional details**

- ExifReader version: 4.23.7
- Node version: 20.3.0

**How to reproduce**

When creating a custom build using this `include` in the package.json:

```json
  "exifreader": {
    "include": {
      "webp": true,
      "png": true,
      "jpeg": true,
      "thumbnail": true,
      "exif": [
        "GPSLatitude",
        "GPSLatitudeRef",
        "GPSLongitude",
        "GPSLongitudeRef",
        "GPSAltitude",
        "GPSAltitudeRef",
        "Thumbnail",
      ]
    }
  }
```

**What I expected would happen:**

One expects to get all specified fields and proper Thumbnail tag

**What really happened:**

Instead, I got this in Thumbnails:

```js
{
   "Thumbnail":  {
      "undefined-259": {"id": 259, "value": 6, "description": 6},
      "undefined-282": {"id": 282, "value": Array(2), "description": Array(2)},
      "undefined-283": {"id": 283, "value": Array(2), "description": Array(2)},
      "undefined-296": {"id": 296, "value": 2, "description": 2},
      "undefined-513": {"id": 513, "value": 20468, "description": 20468},
      "undefined-514": {"id": 514, "value": 5363, "description": 5363}
   }
}
```

**Comment by Jonarod** [comment]
I solved this by including all expected Thumbnail tags, like this:

```js
  "exifreader": {
    "include": {
      "webp": true,
      "png": true,
      "jpeg": true,
      "thumbnail": true,
      "exif": [
        "ColorSpace",
        "DateTime",
        "GPSLatitude",
        "GPSLatitudeRef",
        "GPSLongitude",
        "GPSLongitudeRef",
        "GPSAltitude",
        "GPSAltitudeRef",
        "ExposureProgram",
        "ExposureTime",
        "ISOSpeedRatings",
        "Make",
        "Model",
        "FNumber",
        "ImageWidth",
        "ImageLength",
        "FocalLength",
        "PixelXDimension",
        "PixelYDimension",
        "Orientation",
        "FileType",
        "Thumbnail",

         // added the tags below
        "Compression",
        "JPEGInterchangeFormat",
        "JPEGInterchangeFormatLength",
        "ResolutionUnit",
        "XResolution",
        "YResolution",
        "base64",
        "image",
        "type"
      ]
    }
  }
```

Feel free to close the issue if this is the intended behaviour, but I feel the docs are misleading:


`thumbnail` | Thumbnail. Needs exif.

I guess maybe something like this would be better:

`thumbnail` | Thumbnail. Needs at least `Compression`, `JPEGInterchangeFormat`, `JPEGInterchangeFormatLength`, `ResolutionUnit`, `XResolution`, `YResolution`, `base64`, `image` and `type`

**Comment by mattiasw** [comment]
Setting `"exif": true` should be enough. That's what is meant in the README. And if one wants to be more specific one could do as you did. Would it be clearer if it said something like this?
```
`thumbnail` | Thumbnail. Needs `exif` to be either `true` or an array of the tag names you want from the thumbnail.
```

**Comment by Jonarod** [comment]
What I mean is that just from the README, one would not know that we actually need to specify each and every subfields of the "Thumbnail" field to get the Thumbnail information. Just putting `thumbnail: true` is not enough, and putting "Thumbnail" as one of the desired exif fields is not enough either. We explicitly need to put this: 

Compression, JPEGInterchangeFormat, JPEGInterchangeFormatLength, ResolutionUnit, XResolution, YResolution, base64, image and type


Missing one of these will return gibberish fields inside "Thumbnails".

**Comment by mattiasw** [comment]
I tried setting `"exif": true` and it works fine for me. No need to specify every tag. But if you want the thumbnail image while also not including _all_ the Exif tags, `JPEGInterchangeFormat` and `JPEGInterchangeFormatLength` are the required ones. I will see if I can automate this so you don't have to think about it. The others you mention are only if you want to read those tags from the thumbnail's Exif data, same as for the main image. `type`, `image`, and `base64` are not real tags and should not be needed, they will always be present (those are what `"thumbnail": true` enables).

**Comment by Jonarod** [comment]
Yes when you just put `exif: true`, I understand it automatically put all the exif fields for you. What I meant is when you are selective in your Exif fields, using the `include` and you manually list some Exif tags you need, for example if I only want the GPS, Date and image dimensions related tags but I also happen to need the Thumbnail. In that specific case, again when manually mentioning tags to include in my build, I initially thought that adding either of the "Thumbnail" tag and/or `thumbnail: true` would just work out of the box. There was no mention of the need to add any of the sub-tags I mentioned.

Yeah I guess your solution to auto add all tags when using `thumbnail: true` is more intuitive. I'd say the perfect solution would be that `thumbnail: true` should force all Thumbnails tags, and those who don't want all the tags may just avoid `thumbnail: true` but manually list each tag they actually want as an EXIF `include` list.

**Comment by mattiasw** [comment]
A fix to automatically enable the thumbnail when setting `"thumbnail": true` has been released as version 4.24.0.
